core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -518,6 +548,35 @@ private ExprValue exprFromDays(ExprValue exprValue) {
     return new ExprDateValue(LocalDate.ofEpochDay(exprValue.longValue() - DAYS_0000_TO_1970));
   }
 
+  private ExprValue exprFromUnixTime(ExprValue time) {
+    if (0 > time.doubleValue()) {
+      return ExprNullValue.of();
+    }
+    // According to MySQL documentation:
+    //     effective maximum is 32536771199.999999, which returns '3001-01-18 23:59:59.999999' UTC.
+    //     Regardless of platform or version, a greater value for first argument than the effective
+    //     maximum returns 0.
+    if (32536771200d <= time.doubleValue()) {
+      return ExprNullValue.of();
+    }
+    return new ExprDatetimeValue(exprFromUnixTimeImpl(time));
+  }
+
+  private LocalDateTime exprFromUnixTimeImpl(ExprValue time) {
+    return LocalDateTime.ofInstant(
+            Instant.ofEpochSecond((long)Math.floor(time.doubleValue())),
+            ZoneId.of("UTC"))
+        .withNano((int)((time.doubleValue() % 1) * 1E9));
```

Comment:
its kind of funny how you need to do this

Reply:
A lot of headache with fractional second part